core/src/main/java/io/aiven/klaw/service/UsersTeamsControllerService.java

```diff
@@ -954,4 +1015,68 @@ private boolean userNamePatternValidation(String userName) {
     Matcher m2 = defaultPattern.matcher(userName);
     return m1.matches() || m2.matches();
   }
+
+  public List<TeamModel> getSwitchTeams(String userId) {
+    List<TeamModel> teamModelList = new ArrayList<>();
+    UserInfo userInfo = manageDatabase.getHandleDbRequests().getUsersInfo(userId);
+    int tenantId = commonUtilsService.getTenantId(getUserName());
+    if (userInfo.isSwitchTeams()) {
+      try {
+        String switchAllowedTeamIds = userInfo.getSwitchAllowedTeamIds();
+        if (switchAllowedTeamIds != null) {
+          Set<Integer> teamIds =
+              OBJECT_MAPPER.readValue(switchAllowedTeamIds, new TypeReference<>() {});
+          for (Integer switchAllowedTeamId : teamIds) {
```

Comment:
To prevent you from having to process these values across I think you could set the DB column to Set<Integer> and then write it out as a string using a converter e.g. this is one I'm currently using in my branch for creating an env mapping.


@Converter
public class EnvTagListConverter implements AttributeConverter<List<EnvTag>, String> {

  ObjectMapper mapper = new ObjectMapper();

  @Override
  public String convertToDatabaseColumn(List<EnvTag> envTags) {
    String envTagsStr = null;
    try {
      envTagsStr = mapper.writeValueAsString(envTags);
    } catch (JsonProcessingException e) {
      log.error("Exception converting object to json: {}", e.getMessage());
    }
    return envTagsStr;
  }

  @Override
  public List<EnvTag> convertToEntityAttribute(String envTagsStr) {
    List<EnvTag> envTags = null;
    try {
      envTags = mapper.readValue(envTagsStr, new TypeReference<List<EnvTag>>() {});
    } catch (JsonProcessingException e) {
      log.error("Exception converting json to object: {}", e.getMessage());
    }
    return envTags;
  }
}